src/iris_grib/tests/unit/load_convert/test_product_definition_template_0.py

```diff
@@ -81,8 +82,14 @@ def test_generating_process_warnings(self):
         emit_warnings = convert_options.warn_on_unsupported
         try:
             convert_options.warn_on_unsupported = True
-            with mock.patch("warnings.warn") as warn:
-                product_definition_template_0(section_4(), metadata, rt_coord)
+            future_kwargs = {}
```

Comment:
I assume you do it this way (via a dictionary) for a reason? I'm guessing it's a best practice that I'm not aware of.

Reply:
Because it lets me always use a context block even when it's not doing anything, 
so that avoids repeating the code which appears inside the context, i.e. the alternative could be like ...
```
  if has_attr(IRIS_FUTURE, "date_microseconds"):
      with IRIS_FUTURE.context(date_microseconds=True):
          product_definition_template_0(section_4(), metadata, rt_coord)
  else:
      product_definition_template_0(section_4(), metadata, rt_coord)
  . . .
```
 -- that's really the only reason, cos I hate repeated code !

Reply:
Oh, fair! Yep, I'm happy with that.